### DISCUSSION TITLE
Columns and number of results are stored for all tables at once

### DISCUSSION THREAD

**Discussion by elephantux** [rappasoft/laravel-livewire-tables#618] [Q&A]
Let's say I have a table with users and a table with admins that are on different pages of the site. The table in the database is the same, different conditions in the query (```where->('is_admin', true)```).

If I uncheck (```public bool $columnSelect = true;```) the name in one table, it will also be unchecked in the other table.

The same goes for the number of results on the page. I select 50 in one table, it will be the same in the other (other page).

How to solve this?

**Comment by rappasoft** [comment]
You need to make two different tables to have different options.

**Comment by elephantux** [comment]
@rappasoft did you mean DB tables or Table Classes (I already use it)?

**Reply by rappasoft** [reply]
Table classes. The options are table-specific, so if you want one to have `$columnSelect = true` and one to be `false`, they need to be separate components.

**Comment by elephantux** [comment]
I have two different classes for these tables. But let's say the email field, if disabled in one table on one page, is also disabled in the other table.

**Reply by rappasoft** [reply]
Disabled how?

**Reply by elephantux** [reply]
Disabled is not quite the right word. I meant if the column is disabled in the table filter ($columnSelect = true).

**Reply by rappasoft** [reply]
If they are separate components with different names then they should have different state.
